clientconn.go

```diff
@@ -275,7 +285,21 @@ func DialContext(ctx context.Context, target string, opts ...DialOption) (conn *
 	for _, opt := range opts {
 		opt(&cc.dopts)
 	}
-
+	if cc.dopts.timeout > 0 {
+		ctx, _ = context.WithTimeout(ctx, cc.dopts.timeout)
```

Comment:
you should never ignore the `cancel` function this way - `defer cancel()` on the next line instead.